src/RA_Dlg_GameLibrary.cpp

```diff
@@ -260,36 +260,30 @@ void Dlg_GameLibrary::ThreadedScanProc()
 
     while (FilesToScan.size() > 0)
     {
-        mtx.lock();
-        if (!Dlg_GameLibrary::ThreadProcessingAllowed)
         {
-            mtx.unlock();
-            break;
+            std::scoped_lock lock{mtx};
+            if (!Dlg_GameLibrary::ThreadProcessingAllowed)
+                break;
         }
-        mtx.unlock();
 
-        FILE* pf = nullptr;
-        if ((fopen_s(&pf, FilesToScan.front().c_str(), "rb") == 0) && pf)
+        std::basic_ifstream<unsigned char> ifile{FilesToScan.front(), std::ios::binary};
+        if (ifile.is_open())
         {
             // obtain file size:
-            fseek(pf, 0, SEEK_END);
-            const DWORD nSize = ftell(pf);
-            rewind(pf);
+            const auto nSize = ra::filesize(FilesToScan.front());
 
             auto pBuf = std::make_unique<unsigned char[]>(6 * 1024 * 1024);
 
-            fread(static_cast<void*>(pBuf.get()), sizeof(unsigned char), nSize, pf); // Check
-            Results.insert_or_assign(FilesToScan.front(), RAGenerateMD5(pBuf.get(), nSize));
-            pBuf.reset();
+            ifile.read(pBuf.get(), nSize); // Check
+            Results.insert_or_assign(FilesToScan.front(),
+                                     RAGenerateMD5(std::move(pBuf), gsl::narrow<std::size_t>(nSize)));
```

Comment:
The analyzer didn't like `unique_ptr` because it didn't use ownership semantics, I overloaded `RAGenerateMD5 `to give it ownership semantics.